### PR TITLE
Implement more of drawable

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/game/client/draw/drawable.cpp [hunk 1]
```diff
@@ -178,21 +181,58 @@ ClientUpdateModule *Drawable::Find_Client_Update_Module(NameKeyType key)
...
+
+                max_bones -= val;
+            }
+        }
+    }
+
+    return ret;
 }
 
 bool Drawable::Get_Should_Animate(bool should) const
```
Comment (xezon) [inline src/game/client/draw/drawable.cpp]: `bool when_powered`

File: src/game/client/draw/drawable.cpp [hunk 2]
```diff
@@ -178,21 +181,58 @@ ClientUpdateModule *Drawable::Find_Client_Update_Module(NameKeyType key)
...
+    if (!object->Is_KindOf(KINDOF_PRODUCED_AT_HELIPAD)
+        && (object->Get_Disabled_State(DISABLED_TYPE_DISABLED_HACKED)
+            || object->Get_Disabled_State(DISABLED_TYPE_DISABLED_PARALYZED)
+            || object->Get_Disabled_State(DISABLED_TYPE_DISABLED_EMP)
+            || object->Get_Disabled_State(DISABLED_TYPE_DISABLED_SUBDUED)
+            || object->Get_Disabled_State(DISABLED_TYPE_DISABLED_UNMANNED))) {
+        return false;
+    }
+
+    return !should && !object->Get_Disabled_State(DISABLED_TYPE_DISABLED_UNDERPOWERED);
```
Comment (xezon) [inline src/game/client/draw/drawable.cpp]: Can you make this a condition for `return false` too and return `true` at the end? That way this reads easier.

File: src/game/client/draw/drawable.cpp [hunk 3]
```diff
@@ -1825,8 +1865,232 @@ void Drawable::Set_Animation_Frame(int frame)
...
+
+bool Drawable::Client_Only_Get_First_Render_Obj_Info(Coord3D *position, float *radius, Matrix3D *transform)
+{
+    DrawModule **modules = Get_Draw_Modules();
+    ObjectDrawInterface *draw;
+
+    if (modules != nullptr && *modules != nullptr) {
+        draw = (*modules)->Get_Object_Draw_Interface();
+    } else {
+        draw = 0;
```
Comment (xezon) [inline src/game/client/draw/drawable.cpp]: nullptr

File: src/game/client/draw/drawable.cpp (hunk 1 above)
Comment (jonwil) [reply src/game/client/draw/drawable.cpp]: fixed.

File: src/game/client/draw/drawable.cpp (hunk 2 above)
Comment (jonwil) [reply src/game/client/draw/drawable.cpp]: fixed.

File: src/game/client/draw/drawable.cpp [hunk 4]
```diff
@@ -1825,8 +1869,232 @@ void Drawable::Set_Animation_Frame(int frame)
...
+            case STEALTHLOOK_VISIBLE_FRIENDLY:
+            case STEALTHLOOK_VISIBLE_DETECTED_FRIENDLY: {
+                float opacity = g_theWriteableGlobalData->m_stealthFriendlyOpacity;
+                Object *object = Get_Object();
+
+                if (object == nullptr) {
+                    m_effectiveOpacity1 = opacity;
+                    m_stealthInvisible = false;
+
+                    if (look != STEALTHLOOK_VISIBLE_DETECTED_FRIENDLY || Is_KindOf(KINDOF_MINE)) {
```
Comment (tomsons26) [inline src/game/client/draw/drawable.cpp]: flip for all 3

Comment (jonwil) [reply src/game/client/draw/drawable.cpp]: fixed.

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**